### PR TITLE
Addition of function get_events_detail

### PR DESCRIPTION
**Pull request by rkenning**
As mentioned in Slack, I have created a new function to scrape further details from the Bath Events pages such as Event time, Event location, Location address etc....  Hopefully the returned DF can be of use in providing a richer feature set to any further models.

I have tried to keep the new function code and remarks in keeping with the existing functions in  Web_Scraping.R however if any changes are required please drop me a line.  Also  I have used the library Rvest for my scraping however I was unable to work out how to add it to the library manifest (if that's the correct term)

### CODE REVIEW

**Comment by rkenning** [comment]
As per commit messages above, I identified a bug with the new function when running outside the work space I originally developed in (i.e. the function used a cached work space variable ).

File: r/BANEScarparking/R/web_scraping.R [hunk 1]
```diff
@@ -215,4 +215,113 @@ get_events <- function(from, to) {
...
+#' events <- get_events_detail(DF$year_month_day, DF$timeslot, DF$event_name, DF$event_location_name, DF$event_postcode, DF$event_street, DF$event_locality, DF$event_start, DF$event_end)
+#' @export
+
+get_events_detail <- function(from, to) {
+  # Get all year-month combinations in specified range
+  year_month_day_seq <- seq(as.Date(from), as.Date(to), by = "days") %>%
+    substr(., 0, nchar(.)+6)
+  
+  # Create web addresses for past event pages
+  addresses <- paste0("http://www.bath.co.uk/events/", year_month_day)
```
Comment (kinbiko) [inline r/BANEScarparking/R/web_scraping.R]: looks like the `year_month_day` variable is used before it's declared.

File: r/BANEScarparking/R/web_scraping.R [hunk 2]
```diff
@@ -215,4 +215,113 @@ get_events <- function(from, to) {
...
+        event_postcode[j] <- html_node(event_detail_nodes[j], xpath=".//*[contains(concat( ' ', @class, ' ' ), 'tribe-postal-code')]") %>% html_text_na()
+        event_street[j] <- html_node(event_detail_nodes[j], xpath=".//*[contains(concat( ' ', @class, ' ' ), 'tribe-street-address')]") %>% html_text_na()
+        event_locality[j] <- html_node(event_detail_nodes[j], xpath=".//*[contains(concat( ' ', @class, ' ' ), 'tribe-locality')]") %>% html_text_na()
+        event_start[j] <- html_node(event_detail_nodes[j], xpath=".//*[contains(concat( ' ', @class, ' ' ), 'tribe-event-date-start')]") %>% html_text_na()
+        event_end[j] <- html_node(event_detail_nodes[j], xpath=".//*[contains(concat( ' ', @class, ' ' ), 'tribe-event-date-end')]") %>% html_text_na()
+      }
+      
+        
+      # Make a Dataframe from the populated vectors
+      new_events <- data.frame(year_month_day_seq[ymd],time_slot, event_name, event_location_name, event_postcode, event_street, event_locality, event_start, event_end)
```
Comment (kinbiko) [inline r/BANEScarparking/R/web_scraping.R]: I keep getting  `Error in data.frame(year_month_day_seq[ymd], time_slot, event_name, event_location_name,  : arguments imply differing number of rows: 1, 0` at this point.

I tested this with `get_events_detail("2016-09-09", "2017-02-02")` as my input.

Am I doing something wrong?

**Comment by rkenning** [comment]
Sorry, I got caught out by the Workspace variables again!  (Now I have found the rm(list=ls(all=TRUE))  command this shouldn't happen again) . I have committed a change to my pull request which should resolve the issue.

**Pull request merged**